### PR TITLE
CDK: Add support for input format parsing at jinja macro format_datetime

### PR DESCRIPTION
**Pull request by cmm-airbyte**
## What
In order to support complex input datetime formats at low code, including input_format param to format_datetime  macro, so it will allow us to parse any kind of datetime string, so it can be processed to standarized ISO (or any other format as required).

## Can this PR be safely reverted and rolled back?

- [X] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-cdk/python/pyproject.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ build-backend = "poetry.core.masonry.api"
 
 [tool.poetry]
 name = "airbyte-cdk"
-version = "2.2.0"
+version = "2.2.1"
```
Comment (girarda) [inline airbyte-cdk/python/pyproject.toml:7]: you shouldn't need to bump the version. the github action wil ltake care of it

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/macros.py [hunk 2]
```diff
@@ -107,7 +107,7 @@ def duration(datestring: str) -> Union[datetime.timedelta, isodate.Duration]:
     return parse_duration(datestring)  # type: ignore # mypy thinks this returns Any for some reason
 
 
-def format_datetime(dt: Union[str, datetime.datetime], format: str) -> str:
+def format_datetime(dt: Union[str, datetime.datetime], format: str, input_format: str = None) -> str:
```
Comment (girarda) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/macros.py:110]: nit: input_format is an `Optional[str]`

**Review by girarda — APPROVED** [review]
minor nit. looks good :shipit:

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/macros.py (hunk 2 above)
Comment (cmm-airbyte) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/macros.py]: done

File: airbyte-cdk/python/pyproject.toml (hunk 1 above)
Comment (cmm-airbyte) [reply airbyte-cdk/python/pyproject.toml]: done

**Review by maxi297 — APPROVED** [review]
LGTM! Thanks for addressing the comments!

**Review by aldogonzalez8 — APPROVED** [review]
APPROVED

**Review by bazarnov — APPROVED** [review]

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/macros.py [hunk 3]
```diff
@@ -107,7 +107,7 @@ def duration(datestring: str) -> Union[datetime.timedelta, isodate.Duration]:
     return parse_duration(datestring)  # type: ignore # mypy thinks this returns Any for some reason
 
 
-def format_datetime(dt: Union[str, datetime.datetime], format: str) -> str:
+def format_datetime(dt: Union[str, datetime.datetime], format: str, input_format: Optional[str] = None) -> str:
```
Comment (bazarnov) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/macros.py:110]: There are errors for this test:

```
==================================== ERRORS ====================================
_ ERROR collecting airbyte-cdk/python/unit_tests/sources/declarative/interpolation/test_macros.py _
airbyte-cdk/python/unit_tests/sources/declarative/interpolation/test_macros.py::test_format_datetime: in "parametrize" the number of names (4):
  ['test_name', 'input_value', 'format', 'expected_output']
must be equal to the number of values (5):
  ('test_datetime_string_to_date', '2022-01-01T01:01:01Z', '%Y-%m-%d', None, '2022-01-01')
=============================== warnings summary ===============================
airbyte_cdk/connector_builder/connector_builder_handler.py:28
```

**Review by bazarnov — APPROVED** [review]

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/macros.py (hunk 3 above)
Comment (cmm-airbyte) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/macros.py:110]: done

**Pull request merged**